### ISSUE TITLE
Why is fish shell not POSIX compliant?

### ISSUE THREAD

**Issue by BSiddharth** [fish-shell/fish-shell#10742]
Pretty dumb question, but not being POSIX compliant is on top of the list for anti-fish people. So why is that?

**Comment by faho** [comment]
Because we would like to be free of the limitations and awkward decisions of POSIX.

**Comment by BSiddharth** [comment]
> Because we would like to be free of the limitations and awkward decisions of POSIX.

Can you expand on this? Like what are those awkward decisions or link some resources to read from.

> If you would like to use a POSIX-compliant shell there are many to choose from, so pick one.

I don't care, I just use a different shell whenever needed, so it's not really up for discussion.

**Comment by faho** [comment]
> Can you expand on this? Like what are those awkward decisions or link some resources to read from.

A non-exhaustive list:

- Word-splitting. It's the reason you hear about problems about "filenames with spaces" - a constant `$foo` vs `"$foo"`
- An overreliance on sigils - `$#` vs `$*` vs `$@` vs `$?` vs `$$`, `$()` vs `$(())` vs `<()`
- Awkward block-ending and starting syntax - `if foo; then; fi` vs `case; ...; esac` vs `while foo; then; done` (not `elihw`)
- Awkward tacked-on extensions - `${foo[@]}` for arrays
- Terrible defaults - `read` needs to be used as `IFS= read -r foo` to not break
- A lot of deprecations that were never removed - \`foo\` backtick syntax is still there for people to trip over
- Weird shortcuts that mean more to learn: `for arg; do foo; done` exists and goes over the positional arguments
- A ton of workarounds for missing extensions and underspecified features if you really want to handle multiple shells

Basically read https://mywiki.wooledge.org/BashPitfalls and https://mywiki.wooledge.org/BashFAQ and see how many of those are a direct result of POSIX script being weird.
